### PR TITLE
More loader work

### PR DESCRIPTION
**Pull request by jonallured**
This PR represents a bunch of work around understanding how loaders glue together and also are tested. I naively thought "oh, I'll just go through and find all the places where we're not using already extracted loaders and update them" - that's a lot of places!!

So, that's where I started, but then I found that I didn't always understand how the tests worked, so got into some trouble there and had to fix some things there. This was quite interesting and exposed me to some patterns there that I hadn't seen.

Anyway, time to open a PR for this and then move onto something else. 😉

### CODE REVIEW

File: schema/home/context.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { create, assign } from "lodash"
 import gravity from "lib/loaders/gravity"
+import artistLoader from "lib/loaders/per_type"
```
Comment (alloy) [inline schema/home/context.js]: This isn’t correct and would result in a singleton that never releases memory, kinda like the `gravity` Singleton we’re moving away from. Instead accept more arguments like all other `resolve` functions you’ve dealt with. The 4th one is the rootValue where you can extract the instance of the loader specific to the request.

File: schema/home/fetch.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
 import gravity from "lib/loaders/gravity"
+import fairsLoader from "lib/loaders/per_type"
```
Comment (alloy) [inline schema/home/fetch.js]: Idem ditto.

File: schema/home/title.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { featuredAuction, featuredFair, featuredGene } from "./fetch"
-import gravity from "lib/loaders/gravity"
+import artistLoader from "lib/loaders/per_type"
```
Comment (alloy) [inline schema/home/title.js]: Idem ditto

File: schema/home/context.js (hunk 1 above)
Comment (jonallured) [inline schema/home/context.js]: Addressed with 891550d. I'm confused about why/how one is able to take in these four arguments in seemingly all situations. Is it like global context or something?

Comment (alloy) [inline schema/home/context.js]: Here’s some details about the signature of resolver functions http://dev.apollodata.com/tools/graphql-tools/resolvers.html#Resolver-function-signature. Interesting to note is that supposedly the third parameter is supposed to be used for context such as DataLoader instances, so maybe we’ve been using the wrong parameter (4th) this whole time 🤔

If you want to take a look into that, please go for it.

**Comment by alloy** [comment]
Noice 👌

**Comment by damassi** [comment]
Super nice first Metaphysics PR @jonallured

**Comment by jonallured** [comment]
Yeah, except for the whole thing about breaking staging! 😝

**Pull request merged**